src/motion_input_shaping/zero_vibration_stream_generator.py

```diff
@@ -0,0 +1,361 @@
+        )
+
+        # Record current values and previous positions for next step
+        previous_position = current_position
+        previous_velocity = current_velocity
+
+    return stream_segments
+
+
+class ZeroVibrationStreamGenerator:
+    """A class for creating stream motion with zero vibration input shaping theory."""
+
+    # pylint: disable=too-many-instance-attributes
+    # Ignore number of attributes warning because calling property setters in init is being counted
+    # as additional attributes
+
+    def __init__(
+        self,
+        resonant_frequency: float,
+        damping_ratio: float,
+        shaper_type: ShaperType = ShaperType.ZV,
+    ) -> None:
+        """
+        Initialize the class.
+
+        :param resonant_frequency: The target vibration resonant frequency in Hz.
+        :param damping_ratio: The target vibration damping ratio.
+        :param damping_ratio: Minimum timestep in pvt sequence in seconds.
+        """
+        self.resonant_frequency = resonant_frequency
+        self.damping_ratio = damping_ratio
+        self.shaper_type = shaper_type
+
+        # Initialize impulses as a single impulse with no delay for no shaping
+        self._impulse_times = [0.0]
+        self._impulses = [1.0]
+        self._impulses_updated = False
+        self._update_shaper_impulses()  # Update impulses
+
+    @property
+    def resonant_frequency(self) -> float:
+        """Get the target resonant frequency for input shaping in Hz."""
+        return self._resonant_frequency
+
+    @resonant_frequency.setter
+    def resonant_frequency(self, value: float) -> None:
+        """Set the target resonant frequency for input shaping in Hz."""
+        if value <= 0:
+            raise ValueError(f"Invalid resonant frequency: {value}. Value must be greater than 0.")
+        self._resonant_frequency = value
+        self._impulses_updated = False
+
+    @property
+    def resonant_period(self) -> float:
+        """Get the target resonant period for input shaping in s."""
+        return 1.0 / self.resonant_frequency
+
+    @property
+    def damping_ratio(self) -> float:
+        """Get the target damping ratio for input shaping."""
+        return self._damping_ratio
+
+    @damping_ratio.setter
+    def damping_ratio(self, value: float) -> None:
+        """Set the target damping ratio for input shaping."""
+        if value < 0:
+            raise ValueError(
+                f"Invalid damping ratio: {value}. Value must be greater than or equal to 0."
+            )
+        self._damping_ratio = value
+        self._impulses_updated = False
+
+    @property
+    def shaper_type(self) -> ShaperType:
+        """Get input shaper type."""
+        return self._shaper_type
+
+    @shaper_type.setter
+    def shaper_type(self, value: ShaperType) -> None:
+        """Set input shaper type."""
+        self._shaper_type = value
+        self._impulses_updated = False
+
+    def get_impulses(self) -> list[float]:
```

Comment:
This method name is not the same as the non-stream shaper. It should be for consistency. Change one or the other.